src/parser/postgresparser.cpp

```diff
@@ -70,112 +72,108 @@ std::vector<std::unique_ptr<SQLStatement>> PostgresParser::BuildParseTree(const
 
   pg_query_parse_finish(ctx);
   pg_query_free_parse_result(result);
-  return transform_result;
+  return parse_result;
 }
 
-std::vector<std::unique_ptr<SQLStatement>> PostgresParser::ListTransform(List *root) {
-  std::vector<std::unique_ptr<SQLStatement>> result;
-
+void PostgresParser::ListTransform(ParseResult *parse_result, List *root) {
   if (root != nullptr) {
     for (auto cell = root->head; cell != nullptr; cell = cell->next) {
       auto node = static_cast<Node *>(cell->data.ptr_value);
-      result.emplace_back(NodeTransform(node));
+      parse_result->AddStatement(NodeTransform(parse_result, node));
```

Comment:
This seems kind of iffy to me that you have to call AddStatement on parse result and you're also passing parse_result on NodeTransform. In some cases NodeTransform seems to directly modify parse_result, such as in the cases where WhereTransform is called, the WhereTransform calls `parse_result->AddExpression(std::move(expr));` So why not do something similar and let the Transforms add the statement to parse_result when necessary?

Reply:
I vaguely remember wanting a convention of, "automatically updates the parse result with non-owned stuff, but always return the top-level thing".

I believe there are some instances in the code that need the result of a NodeTransform, to _not_ actually be in the parse result but for some further "stuff it into some other class". For example, PrepareTransform.